src/mlpack/core/optimizers/adagrad/ada_grad.hpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * @file ada_grad.hpp
+ * @author Abhinav Moudgil
+ *
+ * Implementation of the Adagrad optimizer. Adagrad is an optimizer that
+ * chooses learning rate dynamically by adapting to the data. Hence Adagrad
+ * eliminates the need to manually tune the learning rate.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef __MLPACK_CORE_OPTIMIZERS_ADAGRAD_ADA_GRAD_HPP
+#define __MLPACK_CORE_OPTIMIZERS_ADAGRAD_ADA_GRAD_HPP
```

Comment:
As an additional thought, the directory is named `adagrad` but the file is named `ada_grad.hpp`; I think it would be better if we settled on one or the other.  I don't have an opinion on which is better, I'm mostly concerned that we're consistent, is all.